src/MemoryGraph/graph.cs

```diff
@@ -141,6 +141,10 @@ public virtual NodeType AllocTypeNodeStorage()
         /// </summary>
         public int TotalNumberOfReferences { get { return m_totalRefs; } }
 
+        protected int m_segmentSize;
+
+        protected bool m_isUsingSegmentedList = false;
```

Comment:
💡 Consider inlining this as `true`.

Reply:
If we only use one stream class and one internal data structure this flag will be removed.